### PR TITLE
Initial files for app + docker-compose cluster

### PR DESCRIPTION
**Pull request by joe94**
In this PR, I'm adding:
- Code and config files for the hecate app
- Sample docker-compose to show hecate in action syncing webhooks between SNS and Argus.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,12 +2,12 @@ DEFAULT: build
...
 GO           ?= go
 GOFMT        ?= $(GO)fmt
-APP          := hecate 
+APP          := hecate
 DOCKER_ORG   := xmidt
 
 VERSION ?= $(shell git describe --tag --always --dirty)
 PROGVER ?= $(shell git describe --tags `git rev-list --tags --max-count=1` | tail -1 | sed 's/v\(.*\)/\1/')
-BUILDTIME = $(shell date -u '+%Y-%m-%d %H:%M:%S')
+BUILDTIME = $(shell date -u '+%c')
```
Comment (kristinaspring) [inline Makefile:10]: Does this need to be updated for every application?

File: deploy/docker-compose/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+Suppose we have a XMiDT cluster in which Tr1d1um servers are inserting webhooks to SNS and Caduceus machines are fetching them (both leveraging SNS). Our goal is for the cluster to continue normal operation during the update of services (in this process, different versions of the same service will rely on different backends). Although there are multiple ways to proceed with the migration, we recommend the following:
+
+- **Deploy cluster of hecate machines:** ensure that all the SNS webhook updates are being pushed to Argus so that the new Caduceus machines will function properly.
+- **Perform a rolling update of Caduceus servers:** if hecate is working as expected, there should not be surprises here. By the end of this process, XMiDT should no longer be reading updates from SNS directly (only through Hecate).
+- **Perform a rolling update of Tr1d1um servers:** during this process, webhook writes will come from both SNS-Tr1d1um and Argus-Tr1d1um servers. By the end of this, we should be fully transitioned into a cluster relying only on Argus for webhooks.
+
+## A simple Hecate cluster
+
+In this docker-compose cluster, we willl have:
```
Comment (kristinaspring) [inline deploy/docker-compose/README.md:15]:
```suggestion
In this docker-compose cluster, we will have:
```

File: main.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	GitCommit = "undefined"
+	Version   = "undefined"
+	BuildTime = "undefined"
+)
+
+type Config struct {
+	Webhook xwebhook.Config
+}
+
+func hecate(arguments []string) int {
```
Comment (kristinaspring) [inline main.go:43]: We don't want to use uber fx for this application? I thought the goal was to use it for any new applications moving forward?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+	if err != nil {
+		if logger != nil {
+			logging.Error(logger, emperror.Context(err)...).Log(logging.ErrorKey(), err.Error())
+		}
+		fmt.Fprintf(os.Stderr, "Error: %#v\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+func toNewWebhook(w *webhook.W) (*xwebhook.Webhook, error) {
```
Comment (j-mai) [inline main.go:188]: Just curious - why are you converting webhooks to a different kind of webhook here?

Comment (joe94) [reply main.go:188]: The structs are identical as JSON but not as Go structs

File: main.go (hunk 3 above)
Comment (joe94) [reply main.go:43]: We should if we can! I just didn't make time to explore how nicely the old webhook library would play with fx. @j-mai has some free cycles now and she might give it a try! 🚀

**Review by kristinaspring — APPROVED** [review]

File: main.go (hunk 4 above)
Comment (joe94) [reply main.go:185]: Also, I wanted to drop any dependencies on  the `webhook` lib from `xwebhook`

**Pull request merged**